Update dependencies and tweak .gitignore

cch1 opened the pull request:
Update all dependencies to current versions and overhaul exclusions to ensure that dependencies are pedantically specified.

Tests with versions of Clojure older than 1.8 do *not* pass.  For versions prior to 1.7, the dependency clashes seemed substantial.  For version 1.7.0, the failure is due to the inability to compile the updated Clojurescript dependency.

The Clojurescript dependency was very old.  Given the pace of Clojurescript evolution since the last release of liberator, I think a hard choice needs to be made for future releases:
 
- Support old versions of Clojure but with poor Clojurescript support
- Drop support for old versions of Clojure and add support for current versions of Clojurescript.

ordnungswidrig commented:
Thanks for the solid patch. Clojurescript is only used for the demos. I think it's reasonable to bump clojure to 1.7 and exclude clojurescript for everything before 1.8. I want to get rid of the clojurescript dependency even for the demos eventually.

cch1 commented:
I've reworked this PR to incorporate your input (Clojure 1.7 and 1.8 only, moderately updated Clojurescript compatible with both).  I've also rebased it on top of #271 .  I'm waiting to see it update here ... it should only have two commits now.

cch1 commented:
Any update on this?  I believe the Travis CI report shown in GitHub is wrong -following the link shows that the latest build passed.

cch1 commented:
Hello?

ordnungswidrig commented:
Hey Chris, I just triggered the ci build again. Sorry for not being responsive, I'm planning to focus on liberator maintenance this friday.

ordnungswidrig commented:
The CI build fails on openjdk 6. Not exactly sure what's happening but I'm tempted to drop support for java6 in liberator anyways as AFAIK even java 7 has reached it's end of life.

ordnungswidrig commented:
PR merged. Thanks for the patch!
